macOS support (#32)
craftablescience/ChiraEngine: README.md

## Features
To summarize the major features:

> - Native Windows Linux, and macOS support (OpenGL 4.1)

Reviewer: ```suggestion
- Native Windows, Linux, and macOS support (OpenGL 4.1)
```

## Supported Compilers

> | Compiler (Latest) | Windows            | Linux     | macOS                      |
> |-------------------|--------------------|-----------|----------------------------|
> | GCC               | Supported (MinGW)\* | Supported | Unsupported                |
> | Clang             | Unsupported        | Supported | Supported (Homebrew)\*\*|
> | MSVC              | Supported          | N/A       | N/A                        |
> | Clang-CL          | Supported          | N/A       | N/A                        |

Reviewer: ```suggestion
| Compiler (Latest) | Windows             | Linux     | macOS                    |
|-------------------|---------------------|-----------|--------------------------|
| GCC               | Supported (MinGW)\* | Supported | Unsupported              |
| Clang             | Unsupported         | Supported | Supported (Homebrew)\*\* |
| MSVC              | Supported           | N/A       | N/A                      |
| Clang-CL          | Supported           | N/A       | N/A                      |
```

(\*) MinGW builds on Windows will work, but running the application outside your IDE will require you to copy some DLLs
next to the executable. Check the GitHub Actions script to see which DLLs need copied from the MinGW bin directory.

> (\*\*) Homebrew LLVM/Clang is required because Apple Clang as of this time does not support certain C++ features that are used inside ChiraEngine

Reviewer: ```suggestion
(\*\*) Homebrew LLVM/Clang is required because Apple Clang as of this time does not support certain C++ features that are used inside the engine.
```
